ethexe/observer/src/lib.rs

```diff
@@ -75,20 +74,28 @@ pub enum ObserverEvent {
         timestamp: u64,
         code: Vec<u8>,
     },
-    Block(BlockData),
+    Block(SimpleBlockData),
+    BlockSynced(H256),
 }
 
+// TODO (gsobol): make tests for observer service
 pub struct ObserverService {
-    blobs: Arc<dyn BlobReader>,
     provider: Provider,
+    database: Box<dyn OnChainStorage>,
+    // TODO (gsobol): consider to make clone boxed for BlobRead, in order to avoid Arc usage.
+    blobs_reader: Arc<dyn BlobReader>,
     subscription: Subscription<Header>,
 
-    router: Address,
+    router_address: Address,
```

Comment:
mb flatten them into some RuntimeConfig?

Reply:
You mean this two addresses only?

Reply:
addresses + depth params

Reply:
Done